### PR TITLE
Add list support for `lerpColor` like other color functions

### LINKED ISSUE #6953: lerpColor doesn't support list based colors despite other functions supporting it
### Increasing access

Adding list support for lerpColor would help it to better fit in with the other functions as well as increase the ease of use when interacting with other libraries or dealing with list based colors in general (list based colors are often in of themselves easier to interact with, especially considering how easy they are to use as constants).

### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [X] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [ ] Other (specify if possible)

### Feature enhancement details

Add list support for lerpColor

### PR DESCRIPTION
**Pull request by RandomGamingDev**
Resolves #6953

 Changes:

Adds list support to the `lerpColor` like other color functions. 
(I understand why this function doesn't support string based colors since you should know the actual value of the color when lerping between them, but adding list support is still definitely something that would be useful as seen in the referenced issue. However, if anyone thinks that adding string based color support to `lerpColor` as well would be preferred I'll add that to the PR.)

 Screenshots of the change:

```js
const red = [255, 0, 0];
const blue = [0, 255, 0];

function setup() {
  // These always worked
  fill(red);
  fill(blue);
  // This works now!
  lerpColor(red, blue, 0.5);
}
```

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

### CODE REVIEW

**Comment by davepagurek** [comment]
Thanks, looks good! 

> I understand why this function doesn't support string based colors

Do you mean this as in, we don't *want* to support this, or the `color()` function doesn't handle it? Just looking at your code, it seems like the code may actually already support string based colors. I haven't tested this, but if it does work, I'm ok adding `|String` to the types (and maybe mentioning in the docs that any argument to `color()` also works as an input) if you agree.

**Comment by RandomGamingDev** [comment]
> Thanks, looks good!
> 
> > I understand why this function doesn't support string based colors
> 
> Do you mean this as in, we don't _want_ to support this, or the `color()` function doesn't handle it? Just looking at your code, it seems like the code may actually already support string based colors. I haven't tested this, but if it does work, I'm ok adding `|String` to the types (and maybe mentioning in the docs that any argument to `color()` also works as an input) if you agree.

I mean this as in I understand why you wouldn't support it, but I personally would prefer if it were supported. I'd be fine with adding the `String` type as well and yes, the function already does support String types.

**Comment by ashish1729** [comment]
although I can see that heavy lifting is being done by the Color constructor underneath, will it still be valuable to add examples and unit tests for this update in behavior?

**Comment by RandomGamingDev** [comment]
> although I can see that heavy lifting is being done by the Color constructor underneath, will it still be valuable to add examples and unit tests for this update in behavior?

Probably not. It's pretty expected behavior.

**Comment by davepagurek** [comment]
Agreed that it's probably OK to leave it as is. Maybe we can just add `|String` to the types and then merge?

**Comment by RandomGamingDev** [comment]
> Agreed that it's probably OK to leave it as is. Maybe we can just add `|String` to the types and then merge?

Should we add lists as well and all other types that can be converted into p5.Color? Because for `background()` for instance, it says "p5.Color: any value created by the [color()](https://p5js.org/reference/p5/background/#/p5/color) function" and doesn't list out all the types that can be converted into `p5.Color`, and including all those types could actually lead to more confusion. I think we should convert it to be the same as the rest of the documentation that involves colors like this and copy it from `background()`.

**Comment by RandomGamingDev** [comment]
I've added that instead for now to correspond better to the rest of p5.js's documentation instead of include all types (that would create inconsistencies although it could be nicer for some stuff).

**Comment by davepagurek** [comment]
That works, thanks!

**Review by davepagurek — APPROVED** [review]

File: src/color/creating_reading.js [hunk 1]
```diff
@@ -1007,6 +1007,14 @@ p5.prototype.hue = function(c) {
  */
 p5.prototype.lerpColor = function(c1, c2, amt) {
   p5._validateParameters('lerpColor', arguments);
+
+  if (Array.isArray(c1)) {
```
Comment (davepagurek) [inline src/color/creating_reading.js:1011]: oh actually sorry, just noticed, we only convert to a color if it's an array, but I guess this now means we'd accept strings. Maybe these checks should be `!(c1 instanceof p5.Color)`?

**Review by davepagurek — CHANGES_REQUESTED** [review]

File: src/color/creating_reading.js (hunk 1 above)
Comment (RandomGamingDev) [reply src/color/creating_reading.js]: I've converted the checks to `!(c<n> instanceof p5.Color)` to support all color convertible types.

**Review by davepagurek — APPROVED** [review]
thanks!

**Pull request merged**